tests/diagramconfiguration_test.py

```diff
@@ -414,19 +419,132 @@ class A {
     }
     ''')
 
-    callable = file.get_callables()[0]
+    callable = file.callables[0]
     callable_tree_item = CallableTreeItem(callable)
     callable_tree_item.include()
 
-    for child_callable in callable.get_referenced_callables():
+    for child_callable in callable.referenced_callables:
         child_callable_tree_item = CallableTreeItem(child_callable, callable_tree_item)
         child_callable_tree_item.include()
 
     diagram = callable_tree_item.export()
     expected_diagram = '''@startuml
 
+"{0}" -> A: void A()
 "{0}" -> A: void foo()
 
 @enduml'''.format(file_name)
 
     assert diagram == expected_diagram
+
+
+def test__callable_tree_item__export_project_with_delete__export_correct_diagram(directory):
+    file_name = "function_with_constructor.cpp"
+    file = build_index_with_file(directory, file_name, '''
+    void bar();
+
+    class A {
+        A() {}
+
+        void foo() {}
+    }
+
+    void bar() {
+        A* a = new A();
+        delete a;
+    }
+    ''')
+
+    callable = file.callables[0]
+    callable_tree_item = CallableTreeItem(callable)
+    callable_tree_item.include()
+
+    for child_callable in callable.referenced_callables:
+        child_callable_tree_item = CallableTreeItem(child_callable, callable_tree_item)
+        child_callable_tree_item.include()
+
+    diagram = callable_tree_item.export()
+    expected_diagram = '''@startuml
+
+"{0}" -> A: void A()
+"{0}" -> A: <<destroy>>
```

Comment:
Because it has the CursorKind `CXX_NEW_EXPR`. I had to manually add `CXX_DELETE_EXPR` to the `_is_a_call` function in order for the `delete` expression to show up. But I don't see the value in adding the `new` expression there too since it's already covered by the constructor.

Reply:
Removed both because their `get_usr` function returns an empty string.